### PR TITLE
Standardize locales and improve locale validation

### PR DESCRIPTION
**Pull request by jadchaar**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [x] 🧪  Added **tests** for changed code.
- [x] 🛠️  All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [x] 🧹  All linting checks **pass** when run locally (run `tox -e lint` or `make lint` to find out!).
- [ ] 📚  Updated **documentation** for changed code.
- [x] ⏩  Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes
Closes: https://github.com/arrow-py/arrow/issues/952

### CODE REVIEW

**Comment by jadchaar** [comment]
@systemcatch @krisfremen do ya'll think I should change the variable that is referenced when we throw errors about locales? E.g. duplicate locale registered in a subclass or if a locale does not exist in a call to `get_locale`? Currently, we use the original locale they pass in like `en_US`, but if we are transforming it into `en-us` for our internal lookup, should we display this in the error message instead?

**Review by krisfremen — COMMENTED** [review]
Since we are doing a bit of sanitizing anyway, I think what we have in the locales should be fine to return in the errors, it'll help to adopt the new codes also.

**Comment by systemcatch** [comment]
I agree with Kris, let's start returning the new codes in errors.

**Comment by jadchaar** [comment]
Fixed @krisfremen @systemcatch!

File: arrow/api.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 @overload
 def get(
     *,
-    locale: str = "en_us",
+    locale: str = "en-us",
```
Comment (krisfremen) [inline arrow/api.py:25]: Should we maybe put this in the constants?

DEFAULT_LOCALE?

**Review by krisfremen — COMMENTED** [review]
One question from first glance, otherwise looking good.

File: README.rst [hunk 2]
```diff
@@ -105,7 +105,7 @@ Example Usage
     >>> local.humanize()
     'an hour ago'
 
-    >>> local.humanize(locale='ko_kr')
+    >>> local.humanize(locale='ko-kr')
     '1시간 전'
```
Comment (krisfremen) [inline README.rst:109]: I think this is not the output currently, since the last changes to the Korean locale by comfuture.

**Review by krisfremen — COMMENTED** [review]
One more thought off the top of my head, but I'll have to double check this.

File: README.rst (hunk 2 above)
Comment (krisfremen) [reply README.rst:109]: Indeed.

```
>>> import arrow
>>> kr = arrow.utcnow().shift(hours=-1)
>>> kr.humanize(locale='ko_kr')
'한시간 전'
```

Comment (jadchaar) [reply README.rst]: Fixed

File: arrow/api.py (hunk 1 above)
Comment (jadchaar) [reply arrow/api.py]: Agreed. Done!

**Review by krisfremen — APPROVED** [review]
Looks good to me. 👍

**Comment by jadchaar** [comment]
Look good to you @systemcatch?

**Comment by systemcatch** [comment]
Yeah Jad looks good to merge.

**Pull request merged**